### ISSUE TITLE
RuntimeError: no suitable implementation for this system

### ISSUE THREAD

**Issue by TaceyWong** [nameko/nameko#533]
While run the "helloworld" example:
```sh
Exception AttributeError: "'_SocketDuckForFd' object has no attribute '_closed'" in <bound method _SocketDuckForFd.__del__ of _SocketDuckForFd:14> ignored
Exception AttributeError: "'_SocketDuckForFd' object has no attribute '_closed'" in <bound method _SocketDuckForFd.__del__ of _SocketDuckForFd:15> ignored
Exception AttributeError: "'_SocketDuckForFd' object has no attribute '_closed'" in <bound method _SocketDuckForFd.__del__ of _SocketDuckForFd:17> ignored
Exception AttributeError: "'_SocketDuckForFd' object has no attribute '_closed'" in <bound method _SocketDuckForFd.__del__ of _SocketDuckForFd:18> ignored
Exception AttributeError: "'_SocketDuckForFd' object has no attribute '_closed'" in <bound method _SocketDuckForFd.__del__ of _SocketDuckForFd:8> ignored
Exception AttributeError: "'_SocketDuckForFd' object has no attribute '_closed'" in <bound method _SocketDuckForFd.__del__ of _SocketDuckForFd:9> ignored
starting services: greeting_service
Exception AttributeError: "'_SocketDuckForFd' object has no attribute '_closed'" in <bound method _SocketDuckForFd.__del__ of _SocketDuckForFd:7> ignored
Exception AttributeError: "'_SocketDuckForFd' object has no attribute '_closed'" in <bound method _SocketDuckForFd.__del__ of _SocketDuckForFd:10> ignored
Traceback (most recent call last):
  File "/usr/local/bin/nameko", line 11, in <module>
    sys.exit(main())
  File "/home/tacey/.local/lib/python2.7/site-packages/nameko/cli/main.py", line 66, in main
    args.main(args)
  File "/home/tacey/.local/lib/python2.7/site-packages/nameko/cli/commands.py", line 110, in main
    main(args)
  File "/home/tacey/.local/lib/python2.7/site-packages/nameko/cli/run.py", line 184, in main
    run(services, config, backdoor_port=args.backdoor_port)
  File "/home/tacey/.local/lib/python2.7/site-packages/nameko/cli/run.py", line 131, in run
    service_runner.start()
  File "/home/tacey/.local/lib/python2.7/site-packages/nameko/runners.py", line 66, in start
    SpawningProxy(self.containers).start()
  File "/home/tacey/.local/lib/python2.7/site-packages/nameko/utils/__init__.py", line 186, in spawning_method
    return list(pool.imap(call, self._items))
  File "/home/tacey/.local/lib/python2.7/site-packages/eventlet/greenpool.py", line 175, in imap
    return self.starmap(function, six.moves.zip(*iterables))
  File "/home/tacey/.local/lib/python2.7/site-packages/eventlet/greenpool.py", line 160, in starmap
    eventlet.spawn_n(self._do_map, function, iterable, gi)
  File "/home/tacey/.local/lib/python2.7/site-packages/eventlet/greenthread.py", line 66, in spawn_n
    return _spawn_n(0, func, args, kwargs)[1]
  File "/home/tacey/.local/lib/python2.7/site-packages/eventlet/greenthread.py", line 156, in _spawn_n
    hub = hubs.get_hub()
  File "/home/tacey/.local/lib/python2.7/site-packages/eventlet/hubs/__init__.py", line 116, in get_hub
    use_hub()
  File "/home/tacey/.local/lib/python2.7/site-packages/eventlet/hubs/__init__.py", line 70, in use_hub
    mod = get_default_hub()
  File "/home/tacey/.local/lib/python2.7/site-packages/eventlet/hubs/__init__.py", line 38, in get_default_hub
    import eventlet.hubs.epolls
  File "/home/tacey/.local/lib/python2.7/site-packages/eventlet/hubs/epolls.py", line 27, in <module>
    from eventlet.hubs.hub import BaseHub
  File "/home/tacey/.local/lib/python2.7/site-packages/eventlet/hubs/hub.py", line 23, in <module>
    from eventlet.support import greenlets as greenlet, clear_sys_exc_info, monotonic, six
  File "/home/tacey/.local/lib/python2.7/site-packages/eventlet/support/monotonic.py", line 167, in <module>
    raise RuntimeError('no suitable implementation for this system')
RuntimeError: no suitable implementation for this system
```

**Comment by mattbennett** [comment]
This is the same problem as https://github.com/nameko/nameko/issues/425. 

What versions of Nameko and Eventlet are you using?

**Comment by TaceyWong** [comment]
Thanks for reply !

- nameko-2.8.5

 

- eventlet-0.21.0

  

- greenlet-0.4.13

(on ubuntu 17.10)

**Comment by mattbennett** [comment]
A bug was introduced in Eventlet 0.21.0 that causes this problem on certain operating systems. It is fixed in Eventlet 0.22.0, but unfortunately Nameko is not compatible with Eventlet 0.22.0 yet.

Please pin your eventlet down to 0.20.0. We don't do this in setup.py because this bug only affects Python 2.

**Comment by TaceyWong** [comment]
modify setup.py:`install_requires=["eventlet==0.20.0"]`.It works !
Thanks,it's time to turn to Py3 :smile:
